File: C/24BCSH93/3rd Semester/OS/assignment6/question2.c
/*  Q2. Write a program that will create a child process. The parent has to perform the operation of checking whether the accepted number is even or
        odd, and the child has to check whether the number is prime or not.
*/

#include <stdio.h>
#include <unistd.h>
#include <wait.h>

int is_prime(int);

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);
    int pid = fork();

    if (pid > 0) {
        if (num % 2 == 0) {
            printf("%d is an even number\n", num);
        } else {
            printf("%d is an odd number\n", num);
        }
        wait(NULL);
    } else if (pid == 0) {
        if (is_prime(num)) {
            printf("%d is a prime number\n", num);
        } else {
            printf("%d is not a prime number\n", num);
        }
    } else {
        printf("Fork Failed\n");
    }
    return 0;
}

int is_prime(int num) {
    int i;

    for (i = 2; i <= num / 2; i++) {
        if (num % i == 0) {
            return 0;
        }
    }
    return 1;
}

File: Java/Sec-A/assignment1/question2.java
//  Q2. Write a java program, that will take input as number and check it is even or odd.

import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num = sc.nextInt();

        if (num % 2 == 0) {
            System.out.println(num + " is even number");
        } else {
            System.out.println(num + " is odd number");
        }
    }
}
